### PR TITLE
loader: fix bug that SQL Mode didn't take effect

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
fix bug that loader didn't use SQL Mode when create database connections, this bug is not revealed because `all_mode` has manually set SQL Mode related config

### What is changed and how it works?
if `to` database doesn't have a `sql_mode` item, we use SQL Mode discovered by dumpling's `Init`.

### Check List 

Tests 

 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: loader/loader.go [hunk 1]
```diff
@@ -460,7 +460,20 @@ func (l *Loader) Init(ctx context.Context) (err error) {
...
+	// used to change loader's specified DB settings, currently SQL Mode
+	lcfg, err := l.cfg.Clone()
+	if err != nil {
+		return err
+	}
+	lowerM := map[string]struct{}{}
+	for k := range l.cfg.To.Session {
+		lowerM[strings.ToLower(k)] = struct{}{}
+	}
+	if _, ok := lowerM["sql_mode"]; !ok {
```
Comment (csuzhangxc) [inline loader/loader.go:472]: Is a `bool` value like `hasSQLMode` enough rather than a `map`? (check `strings.ToLower(k) == sql_mode` in the `for` loop)

Comment (lance6716) [reply loader/loader.go]: https://github.com/pingcap/dm/pull/1134/commits/7fb7fdc2aca8695f0878d1ff98d274f1046fb693

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**